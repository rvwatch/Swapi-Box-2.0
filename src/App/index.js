import React, { Component } from 'react';
import Main from '../App/Components/Main';
import './App.css';
import Header from '../App/Components/Header';
import {
  getScrollText,
  getPeopleData,
  getPlanetData,
  getVehicleData,
  getStarshipData
} from '../apiCalls/';

class App extends Component {
  constructor() {
    super();
    this.state = {
      scrollText: {},
      people: [],
      planets: [],
      vehicles: [],
      starships: [],
      favorites: []
    };
  }

  async componentDidMount() {
    const scrollText = await getScrollText('https://swapi.co/api/films/');
    this.setState({ scrollText });
    const people = await getPeopleData('https://swapi.co/api/people');
    this.setState({ people });
    const planets = await getPlanetData('https://swapi.co/api/planets/');
    this.setState({ planets });
    const vehicles = await getVehicleData('https://swapi.co/api/vehicles/');
    this.setState({ vehicles });
    const starships = await getStarshipData('https://swapi.co/api/starships/');
    this.setState({ starships });
  }

  toggleFavorites = selectedCard => {
    const duplicateRemoved = this.state.favorites.filter(
      card => card.id !== selectedCard.id
    );

    const favorites =
      duplicateRemoved.length !== this.state.favorites.length
        ? [...duplicateRemoved]
        : [...this.state.favorites, selectedCard];

    this.setState({ favorites });
  };

  render() {
    return (
      <section className="app-wrap">
        <div className="stars" />
        <div className="twinkling" />
        <Header />
        <Main toggleFavorites={this.toggleFavorites} {...this.state} />
      </section>
    );
  }
}

export default App;
