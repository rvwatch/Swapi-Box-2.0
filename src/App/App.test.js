import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './index';
import {
  mockSelectedCard1,
  mockSelectedCard2,
  mockSelectedCard3,
  mockFavorites
} from '../apiCalls/MockedData';

describe('App ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('toggleFavorites ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should have a default favorites state of an empty array', () => {
    expect(wrapper.state().favorites).toEqual([]);
  });

  it('should add selected cards to favorites', () => {
    wrapper.instance().toggleFavorites(mockSelectedCard1);
    expect(wrapper.state().favorites).toEqual([mockSelectedCard1]);

    wrapper.instance().toggleFavorites(mockSelectedCard2);
    wrapper.instance().toggleFavorites(mockSelectedCard3);
    expect(wrapper.state().favorites).toEqual(mockFavorites);
  });

  it('should remove the card if it is already favorited', () => {
    expect(wrapper.state().favorites).toEqual([]);
    wrapper.instance().toggleFavorites(mockSelectedCard1);
    expect(wrapper.state().favorites).toEqual([mockSelectedCard1]);
    wrapper.instance().toggleFavorites(mockSelectedCard1);
    expect(wrapper.state().favorites).toEqual([]);
  });
});
