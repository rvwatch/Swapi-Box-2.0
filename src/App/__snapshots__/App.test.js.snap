// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stars"
/>,
        <div
          className="twinkling"
/>,
        <Header />,
        <Main
          favorites={Array []}
          people={Array []}
          planets={Array []}
          scrollText={Object {}}
          starships={Array []}
          toggleFavorites={[Function]}
          vehicles={Array []}
/>,
      ],
      "className": "app-wrap",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "twinkling",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favorites": Array [],
          "people": Array [],
          "planets": Array [],
          "scrollText": Object {},
          "starships": Array [],
          "toggleFavorites": [Function],
          "vehicles": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stars"
/>,
          <div
            className="twinkling"
/>,
          <Header />,
          <Main
            favorites={Array []}
            people={Array []}
            planets={Array []}
            scrollText={Object {}}
            starships={Array []}
            toggleFavorites={[Function]}
            vehicles={Array []}
/>,
        ],
        "className": "app-wrap",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "twinkling",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favorites": Array [],
            "people": Array [],
            "planets": Array [],
            "scrollText": Object {},
            "starships": Array [],
            "toggleFavorites": [Function],
            "vehicles": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
